NotesApplication.Data/NotesDbContext.cs

```diff
@@ -31,4 +34,52 @@ public override async ValueTask DisposeAsync()
     {
         await base.DisposeAsync();
     }
+
+    public async Task BeginTransactionAsync(CancellationToken cancellationToken)
+    {
+        _currentTransaction ??= await Database.BeginTransactionAsync(cancellationToken);
+    }
+
+    public async Task CommitTransactionAsync(CancellationToken cancellationToken)
+    {
+        try
+        {
+            await SaveChangesAsync(cancellationToken);
```

Comment:
remove this line because it's redundant.
.SaveChanges already implemented inside handlers